realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -161,10 +176,12 @@ long getNativePtr() {
 
     public void beginTransaction() {
         nativeBeginTransaction(nativePtr);
+        invokeListenerIfSchemaChanged();
     }
 
     public void commitTransaction() {
         nativeCommitTransaction(nativePtr);
+        invokeListenerIfSchemaChanged();
```

Comment:
You are right. WE don't need this if we only support schema changes by remote process.

I added here for supporting the changing the schema version in *this* process.
No need to do this?